Maker-Agents-Suit-Module/Governance-Module/src/lift.events.ts

```diff
@@ -0,0 +1,57 @@
+import { 
+  Finding, 
+  HandleTransaction, 
+  TransactionEvent, 
+  FindingSeverity, 
+  FindingType, 
+  Log,
+} from 'forta-agent';
+import { 
+  AddressVerifier,
+  isAddressKnown,
+  decodeSingleParam,
+} from './utils';
+
+const LIFT_EVENT: string = "0x3c278bd500000000000000000000000000000000000000000000000000000000";
+
+export const createFinding = (alertId: string, unknown: string, topic: number): Finding =>
+  Finding.fromObject({
+    name: "MakerDAO MCDM_ADM lift event detected",
+    description: `Topic #${topic} is an unknown address`,
+    alertId: alertId,
+    type: FindingType.Suspicious,
+    severity: FindingSeverity.High,
+    metadata: {
+      address: unknown,
+    },
+  });
+
+export const provideLiftEventsListener = (
+  alertId: string, 
+  contractAddress: string,
+  isKnown: AddressVerifier = isAddressKnown,
+  topic: string = LIFT_EVENT,
+): HandleTransaction => {
+
+  const contract: string = contractAddress.toLowerCase();
+
+  return async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+
+    if(!txEvent.addresses[contract])
+      return findings;
+
+    for(const log of txEvent.logs) {
+      if((log.address === contract) && (log.topics[0] === topic)){
+        const topic1: string = decodeSingleParam('address', log.topics[1]).toLowerCase();
+        const topic2: string = decodeSingleParam('address', log.topics[2]).toLowerCase();
```

Comment:
Do you need `toLowerCase()` here? AFAIK all the addresses coming from forta-agent types are in lower case. If your `decodeSingleParam` doesn't change the case you don't need it.

Reply:
all the addresses coming from forta-agent types are in lower case, but the decoded address is not lower case, I mean when web3 decode the address it is not returning a lower case one, seems like it returns a checksum address.
Check this code for example:
```javascript
const addr = Web3.utils.leftPad("0xc", 40);
const encoded = web3.eth.abi.encodeParameter('address', addr);
const decoded = web3.eth.abi.decodeParameter('address', encoded);
console.log(decoded);
```
On the other hand if you use `0xd` you will get a lower case address.